tests/swagger/test_resources.py

```diff
@@ -1,104 +1,44 @@
-import json
+import unittest
+import six
 
-from bravado_core.request import unmarshal_request
 from bravado_core.response import validate_response
+from bravado_core.param import marshal_param
 
-from .support import (SwaggerTest, MINIMALIST_BUCKET, MINIMALIST_GROUP,
-                      MINIMALIST_COLLECTION, MINIMALIST_RECORD)
+from .support import SwaggerTest
 
 
-class SwaggerResourcesTest(SwaggerTest):
+class MetaResourcesTest(type):
+    """Metaprogram test cases for easy debugging."""
 
-    allowed_failures = ['version']
+    def __init__(cls, *args, **kargs):
+        unittest.case.TestCase = cls
```

Comment:
This is the weirdest part, I think. We are forcing pytest to look at the Worker class while running this TestCase, without it it would just use the collected version. Whick means it doesn't run any test. This is the main motivation to use a metaclass. See http://doc.pytest.org/en/latest/example/pythoncollection.html and http://stackoverflow.com/questions/2798956/python-unittest-generate-multiple-tests-programmatically